webserial.js

```diff
@@ -96,81 +100,68 @@ class Transport {
             temp_pkt[j] = data[i];
         }
         packet = temp_pkt.slice(0, j); /* Remove unused bytes due to escape seq */
+        this.left_over = new Uint8Array(data).slice(i);
         return packet;
     }
 
     read = async ({timeout=0, min_data=12} = {}) => {
-        let t;
-        let packet = null;
-        let value, done;
         console.log("Read with timeout " + timeout);
-        const reader = this.device.readable.getReader();
-        if (timeout > 0) {
-            t = setTimeout(function() {
-                reader.cancel();
+        let t;
+        let packet = this.left_over;
+        this.left_over = new Uint8Array(0);
+        if (packet.length < min_data) {
+            const reader = this.device.readable.getReader();
+            try {
+                if (timeout > 0) {
+                    t = setTimeout(function() {
+                        reader.cancel();
+                    }, timeout);
+                }
+                do {
+                    const {value, done} = await reader.read();
+                    if (done) {
+                        throw new TimeoutError("Timeout");
+                    }
+                    var p = new Uint8Array(this._appendBuffer(packet.buffer, value.buffer));
+                    packet = p;
+                } while (packet.length < min_data);
+            } finally {
+                if (timeout > 0) {
+                    clearTimeout(t);
+                }
                 reader.releaseLock();
-            }, timeout);
-        }
-
-        do {
-            this.reader = reader;
```

Comment:
You're removing the usage of `this.reader` and always close it in this method. You also need to remove the reference to `this.reader` from `disconnect()`

Reply:
Removed, again, I think that came from another branch. I'll be more carful next time.